apps/server/src/shared/domain/domainobject/user.do.ts

```diff
@@ -0,0 +1,45 @@
+import { BaseWithTimestampsDO, EntityId, LanguageType } from '@shared/domain/index';
+
+export class UserDO extends BaseWithTimestampsDO {
+	email: string;
+
+	firstName: string;
+
+	lastName: string;
+
+	roleIds: EntityId[];
+
+	schoolId: EntityId;
+
+	ldapDn?: string;
+
+	externalId?: string;
+
+	importHash?: string;
+
+	firstNameSearchValues?: string[];
+
+	lastNameSearchValues?: string[];
+
+	emailSearchValues?: string[];
+
+	language?: LanguageType;
+
+	forcePasswordChange?: boolean;
+
+	preferences?: Record<string, unknown>;
+
+	constructor(domainObject: UserDO) {
+		super(domainObject);
+		this.firstName = domainObject.firstName;
+		this.lastName = domainObject.lastName;
+		this.email = domainObject.email;
+		this.schoolId = domainObject.schoolId;
+		this.roleIds = domainObject.roleIds;
+		this.ldapDn = domainObject.ldapDn;
+		this.externalId = domainObject.externalId;
+		this.forcePasswordChange = domainObject.forcePasswordChange;
+		this.language = domainObject.language;
+		this.preferences = domainObject.preferences;
+	}
+}
```

Comment:
For everyone: When I see this, I wonder where is there more value than using entities?

Reply:
Entities are strongly bound to the Database and should not be used in the same manner as DTOs in UCs, which we currently do. It is safer (and less coupled to the DB) to use DOs (Domain Objects) as a way to orchestrate Database Objects and use them safely in our UCs. 

Details for that decision can be found here: https://docs.dbildungscloud.de/x/nwETD

A next step would be to move the repos to fitting UCs. However this discussion is a future topic for the ARC Group.